### DISCUSSION TITLE
Stupid Question: Possible IgnoreMismatchedAddendaCount Flag?

### DISCUSSION THREAD

**Discussion by cchollar** [moov-io/ach#1135] [Q&A] [answered]
Hello! I work for a local bank that partners with various FinTech's to provide them Fed access and regulatory oversight. As a result, we work with multiple financial cores/processors and many of them seem to really struggle with providing actual NACHA compliant ACH files to us. I (being recently tasked with providing reporting of our ach originations) have been experimenting with this wonderful library. However, I ran into an issue that isn't the fault of this parser. You see, one of our processors still doesn't know how to process ALL returns, and upon attempting to parse an ACH origination file with a CTX return, moov-ach refuses to parse the file due to the CTX Return Batch missing its "705" addenda entries, thus causing a mismatch of its actual vs expected addendaCount.

Now I realize the proper solution is to instruct the processor that generated this ACH origination for us to do it right and generate a proper return entry. However, the files I am tasked with parsing reach back 2-3 years, presumably with this issue the entire time, and even future originations will be tainted due to a very slow turnround time. My questions are thus: 

1. Is it stupid to ask for some flag to ignore a Mismatched Addenda Count? (I assume it is)
2. More philosophically: How do I attempt to parse ACH files with bad records if moov-ach cannot be fault tolerant?

I appreciate all the work you developers do working on this project, and I welcome any and all responses from anyone who might see this!

**Comment by adamdecaf** [comment, marked as answer]
Thanks for opening this issue @cchollar! We've seen quite a lot of similar issues as you describe, so I'm happy to help out. 

> 1. Is it stupid to ask for some flag to ignore a Mismatched Addenda Count? (I assume it is)
It's a valid request and we have plenty of [`ValidateOpts` flags](https://pkg.go.dev/github.com/moov-io/ach#ValidateOpts) to cover situations like this.

> 2. More philosophically: How do I attempt to parse ACH files with bad records if moov-ach cannot be fault tolerant?
The ACH library is designed to follow Nacha specifications, guidelines, and rules. We offer knobs/flags to make parsing and validation more flexible.

**Reply by adamdecaf** [reply]
I've opened up https://github.com/moov-io/ach/pull/1137 to add this Validation option.

**Comment by adamdecaf** [comment]
https://github.com/moov-io/ach/releases/tag/v1.24.0 has been released with this. Thanks!

**Reply by cchollar** [reply]
Thank you so much! Really appreciate the work you and the team do on this project
